jaxrs/WFLY-12779_Use-MicroProfile-Config-in-RESTEasy

```diff
@@ -0,0 +1,76 @@
+= Use MicroProfile Config in RESTEasy
+:author:            Ron Sigal
+:email:             rsigal@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Currently, RESTEasy 3.x (targeted to Wildfly) uses servlet context-params to obtain
+configuration information. In RESTEasy 4.x, we have updated configuration to access
+all standard MicroProfile Config sources. Also, we have added three additional ConfigSources:
+
+* servlet init-params [ordinal 60]
+* filter init-params [ordinal 50]
+* servlet context-params [ordinal 40]
+
+All parameters, both RESTEasy parameters and application parameters, are
+available from all ConfigSources to both RESTEasy and application code.
+
+Since this feature will depend on the MicroProfile Config project, it will be 
+included in Wildfly by way of the MicroProfile Extension Pack.
+
+== Issue Metadata
+
+=== Issue
+
+* {issue-base-url}/EAP7-1386 [EAP7-1386]
+* {issue-base-url}/WFLY-12779 [WFLY-12779]
+
+=== Related Issues
+
+* {issue-base-url}/RESTEASY-2406 [RESTEASY-2406]
+
+=== Dev Contacts
+
+* mailto:rsigal@redhat.com[Ron Sigal]
+* mailto:asoldano@redhat.com[Alessio Soldano]
+
+=== QE Contacts
+
+* Petr Kremensky
+
+=== Testing By
+
+[ ] Engineering
+
+[x] QE
+
+=== Affected Projects or Components
+
+* RESTEasy, Wildfly
+
+=== Other Interested Projects
+
+https://github.com/smallrye/smallrye-config
+
+== Requirements
+
+=== Hard Requirements
+
+* Update RESTEasy to retrieve all configuration parameters from MicroProfile ConfigSources.
```

Comment:
Could you please elaborate that this one will apply only in case that MicroProfile classes are available

Reply:
Hi @pkremens,

I've rewritten the proposal to account for the fact that RESTEasy must be able to function with or without the availability of MicroProfile Config.

-Ron

Reply:
Hi @ronsigal 

thanks! It is a public holiday today here in Czech Republic so I'll take a look on Monday.

Reply:
Hi @ronsigal,
thanks for updates, looking good to me. 

I don't think we need to extend the `Test Plan` section of this document more, since there will be a separate document in "jbossqe-eap/test-plans" covering that part (we can mention a new profile `disable.microprofile.tests` also in there, as it will be needed only for EAP testing).